sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/EventProcessorEvent.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.Messaging.EventHubs.Processor
+{
+    /// <summary>
+    ///   A <see cref="PartitionEvent" /> that is strongly tied to an <see cref="EventProcessorClient" />.  It
+    ///   provides a means to create event processing checkpoints.
+    /// </summary>
+    ///
+    public class EventProcessorEvent : PartitionEvent
+    {
+        /// <summary>
+        ///   The <see cref="EventProcessorClient" /> this instance is related to.
+        /// </summary>
+        ///
+        private EventProcessorClient Processor { get; }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="EventProcessorEvent"/> class.
+        /// </summary>
+        ///
+        /// <param name="processor">The <see cref="EventProcessorClient" /> this instance is related to.</param>
+        /// <param name="partitionContext">The context of the Event Hub partition this instance is associated with.</param>
+        /// <param name="eventData">The received event to be processed.  Expected to be <c>null</c> if the receive call has timed out.</param>
+        ///
+        protected internal EventProcessorEvent(EventProcessorClient processor,
```

Comment:
I'd suggest taking a `Func<EventData, Task> and using that for the `UpdateCheckpoint` implementation instead.   No need to hold onto the entire processor here.